files/image_config/monit/container_checker

```diff
@@ -56,6 +56,10 @@ def get_expected_running_containers():
             else:
                 expected_running_containers.add(container_name)
 
+            if container_name == "database" and device_info.is_supervisor():
```

Comment:
IMO, I think it is better to add the `database-chassis` in the feature table for the supervisor card iso for adding the check here

Reply:
Currently, the feature table is init_cfg.json file which is generated in the image building time.  If we want to add the database-chassis docker to the feature table, we have make the init_config.json  as J2 template then allow the the database-chassis container to be inserted/rendered in the feature table for Supervisor card during runtime at system boot up.  Is this what you suggested to do or another method?

Reply:
`init_cfg.json` is generated from init_cfg.json.j2, recently change was done to make `dhcp_relay` feature enabled or disabled based on the attribute in `DEVICE_METADATA`. 
https://github.com/Azure/sonic-buildimage/blob/51a0aed02af5e2586b207b2493de28180dd22bed/files/build_templates/init_cfg.json.j2#L38

I was wondering if we can follow the same approach and enable  `database-chassis` when `switch_type` in DEVICE_METADATA is `fabric`

Reply:
weeks ago, when I looked at this init_cfg.json.j2 with this change, it looks like there is an issue with this change.  During image build,  the init_cfg.json is generated with the following value for the "dhcp-relay".
```
 "dhcp_relay": {
            "state": "{% if not (DEVICE_METADATA is defined and DEVICE_METADATA['localhost'] is defined and DEVICE_METADATA['localhost']['type'] is defined and DEVICE_METADATA['localhost']['type'] != 'ToRRouter') %}enabled{% else %}disabled{% endif %}", false, "enabled")) %}",
            "has_timer" : false,
            "has_global_scope": false,
            "has_per_asic_scope": false,
            "auto_restart": "enabled",
            "high_mem_alert": "disabled"
        },

``` 
And, at boot up for the first time installation, the dhcp-relay state value is no longer rendered to either "enabled" or "disabled" when generated the default configuration.  Based on what I saw, script "sonic-cfggen" will not render init_cfg.json file since it is using option "-j".  if using option "-t" as template, it will work.   Am I missing anything with this change?  Or do you know who renders it before the "sonic-cfggen" combine it to the config_db.json?

Reply:
@arlakshm When the init_cfg,json with value ' "dhcp_relay": {"state": "{% if not (DEVICE_META....." ' is rendered to either "enabled" and "disabled"?  I cannot find where the code renders the value of "state' for dhcp_relay  to "enabled" or "disable".  Please help me understand it and I can make the same design for the database-chassis and test it.  Thanks.

Reply:
@abdosi - will you be able to help answer this? Thanks.

Reply:
After the investigation, I don't think we can use DEVICE_METATDATA|localhost|swtich_type to determine to enable the database-chassis or not.  The dependency of database.service required database-chassis to be start first.  Without start the database-chassis first, 'database' cannot be started and not able to load the DVEVICE_METADATA.
```
In file  database.service

 Wants=database-chassis.service
 After=database-chassis.service
```